Kernel/Net/Routing.h

```diff
@@ -6,25 +6,45 @@
 
 #pragma once
 
+#include <AK/IPv4Address.h>
 #include <Kernel/Locking/MutexProtected.h>
 #include <Kernel/Net/NetworkAdapter.h>
 #include <Kernel/Thread.h>
 
 namespace Kernel {
 
+struct Route : public RefCounted<Route> {
+    Route(IPv4Address const& destination, IPv4Address const& gateway, IPv4Address const& netmask, RefPtr<NetworkAdapter> adapter)
+        : destination(destination)
+        , gateway(gateway)
+        , netmask(netmask)
+        , adapter(move(adapter))
+    {
+    }
+
+    const IPv4Address destination;
+    const IPv4Address gateway;
+    const IPv4Address netmask;
+    RefPtr<NetworkAdapter> adapter;
```

Comment:
This one I was a little iffy about. I used a `RefPtr` because we're passing a `RefPtr<NetworkAdapter>` returned by `NetworkingManagement::the().lookup_by_name`. Thoughts?